src/python/pants/backend/python/lint/pylint/rules_integration_test.py

```diff
@@ -0,0 +1,151 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from functools import partialmethod
+from pathlib import PurePath
+from textwrap import dedent
+from typing import List, Optional
+
+import pytest
+
+from pants.backend.python.lint.pylint.rules import PylintTarget
+from pants.backend.python.lint.pylint.rules import rules as pylint_rules
+from pants.backend.python.targets.python_library import PythonLibrary
+from pants.build_graph.address import Address
+from pants.build_graph.build_file_aliases import BuildFileAliases
+from pants.engine.fs import FileContent, InputFilesContent, Snapshot
+from pants.engine.legacy.structs import PythonTargetAdaptor
+from pants.engine.rules import RootRule
+from pants.engine.selectors import Params
+from pants.rules.core.lint import LintResult
+from pants.source.wrapped_globs import EagerFilesetWithSpec
+from pants.testutil.option.util import create_options_bootstrapper
+from pants.testutil.test_base import TestBase
+
+
+class PylintIntegrationTest(TestBase):
+  source_root = "tests/python/pants_test"
+  good_source = FileContent(path="good.py", content=b"'''docstring'''\nVAR = 42\n")
+  bad_source = FileContent(path="bad.py", content=b"VAR = 42\n")
+
+  create_python_library = partialmethod(
+    TestBase.create_library, path=source_root, target_type="python_library", name="target",
+  )
+
+  def write_file(self, file_content: FileContent) -> None:
+    self.create_file(
+      relpath=str(PurePath(self.source_root, file_content.path)),
+      contents=file_content.content.decode(),
+    )
+
+  @classmethod
+  def alias_groups(cls) -> BuildFileAliases:
+    return BuildFileAliases(targets={'python_library': PythonLibrary})
+
+  @classmethod
+  def rules(cls):
+    return (*super().rules(), *pylint_rules(), RootRule(PylintTarget))
+
+  def run_pylint(
+    self,
+    source_files: List[FileContent],
+    *,
+    config: Optional[str] = None,
+    passthrough_args: Optional[str] = None,
+    interpreter_constraints: Optional[str] = None,
+    skip: bool = False,
+  ) -> LintResult:
+    args = ["--backend-packages2=pants.backend.python.lint.pylint"]
+    if config:
+      # TODO: figure out how to get this file to exist...
```

Comment:
Other reviewers: this is a known TODO introduced from me adding `OptionsBootstrapper` to these integration tests. Iirc correctly, the issue is that `parser.py` does not validate files in the actual build root, only in the cwd. The file exists, but `parser.py` thinks it doesn't.

Reply:
Could we add that relevant background in the text of the comment here, or link to an issue with that info? As it stands this comment doesn't really provide that sufficient context.

Reply:
I opened #9148. Dylan, if you have a moment, it'd be great to link to this ticket here and in the other linter tests.